src/middlewared/middlewared/plugins/jail.py

```diff
@@ -989,6 +1006,65 @@ def rc_action(self, action):
 
         return True
 
+    @accepts(Dict('props', additional_attrs=True))
+    def update_defaults(self, props):
+        iocage = ioc.IOCage(jail='default', reset_cache=True)
+        for prop in props:
+            iocage.set(f'{prop}={props[prop]}')
+
+    @private
+    def retrieve_default_iface(self):
+        default_ifaces = ioc_common.get_host_gateways()
+        if default_ifaces['ipv4']['interface']:
+            return default_ifaces['ipv4']['interface']
+        elif default_ifaces['ipv6']['interface']:
+            return default_ifaces['ipv6']['interface']
+
+    @private
+    def retrieve_vnet_interface(self, iface):
+        if iface == 'auto':
+            return self.retrieve_default_iface()
+        elif iface != 'none':
+            return iface
+        else:
+            return None
+
+    @private
+    def retrieve_vnet_bridge(self, interfaces):
+        bridges = []
+        for interface in interfaces.split(','):
+            if interface.split(':')[-1] not in bridges:
+                bridges.append(interface.split(':')[-1])
+        return bridges
+
+    @private
+    def retrieve_nat_interface(self, iface):
+        return self.retrieve_default_iface() if iface == 'none' else iface
+
+    @private
+    def start_failover_checks(self, jail_config):
+        jail_config = {
+            k: ioc_common.check_truthy(v) if k in IOCJson.truthy_props else v for k, v in jail_config.items()
+        }
+        if jail_config.get('dhcp'):
+            jail_config['vnet'] = 1
```

Comment:
Should we do all that inside the if statement?

Reply:
Well we check if vnet/nat is enabled below, so if we want to do this below, we can use another if clause inside the main one. I'll update this